midas.py

```
@@ -0,0 +1,124 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import create_engine, Column, Integer, String, DateTime, ForeignKey, Table
+from sqlalchemy.orm import sessionmaker, relationship
+from sqlalchemy.ext.declarative.api import DeclarativeMeta
+import sqlalchemy
+
+NAME_LEN = 30
+DEFAULT_VARCHAR_LEN = 256
+
+Base = declarative_base()
+DB_PATH = './midas.db'
+engine = create_engine(f'sqlite:///{DB_PATH}')
+session = sessionmaker(bind=engine)()
+
+
+class Query:
+    """
+    This class is very close to Query class of sqlalchemy, but this class also supports the query API.
+    """
+    def __init__(self, table_type: DeclarativeMeta, query: sqlalchemy.orm.query.Query):
+        """
+        :param table_type: Terrorist or Organization or Event.
+        :param query: Query obbject of sqlalchemy.
+        """
+        self.type = table_type
+        self.query = query
+
+    def __repr__(self):
+        return f'<{self.query.count()} {self.type.__name__}s>'
+
+    def __getattr__(self, item):
+        return getattr(self.query, item)
+
+    def __getitem__(self, item):
+        return self.query.__getitem__(item)
+
+    def refine(self, *entities, **kwargs) -> 'Query':
+        """
+        Same as filter and filter_by of sqlalchemy, but return Query object that supports the query API.
+
+        Example: Terrorist.get().refine(id=111) -> <1 Terrorists>
+        """
+        if kwargs and isinstance(next(iter(kwargs.keys())), str):
```

If you receive both `entities` and `kwargs` you silently ignore the `entities`.